service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsControllerHearingInPastMidEventTest.java

```diff
@@ -0,0 +1,142 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@ActiveProfiles("integration-test")
+@OverrideAutoConfiguration(enabled = true)
+@WebMvcTest(ManageHearingsController.class)
+public class ManageHearingsControllerHearingInPastMidEventTest extends AbstractControllerTest {
+
+    private static final String CASE_ID = "12345";
+
+    @MockBean
+    private FeatureToggleService featureToggleService;
+
+    @MockBean
+    private ManageHearingsService hearingsService;
+
+    @Autowired
+    private Time time;
+
+
+    ManageHearingsControllerHearingInPastMidEventTest() {
+        super("manage-hearings");
+    }
+
+    @Test
+    void shouldPopulateHearingDatesToCorrectOneWhenIncorrectAndToggledOn() {
+        given(featureToggleService.isAddHearingsInPastEnabled()).willReturn(true);
+
+        LocalDateTime correctStartDate = LocalDateTime.of(2050, 6, 5, 11, 0);
+        LocalDateTime correctEndDate = LocalDateTime.of(2050, 6, 5, 11, 0);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(parseLong(CASE_ID))
+            .data(Map.of("hearingStartDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingEndDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingStartDateConfirmation", correctStartDate,
+                "hearingEndDateConfirmation", correctEndDate,
+                "hearingDateConfirmation","No"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "hearing-in-past");
+
+        Map<String, Object> responseData = callbackResponse.getData();
+
+        assertThat(responseData.get("hearingStartDate").equals(correctStartDate));
+        assertThat(responseData.get("hearingEndDate").equals(correctEndDate));
+    }
+
+    @Test
+    void shouldPopulateHearingEndDateToCorrectOneWhenIncorrectAndToggledOn() {
+        given(featureToggleService.isAddHearingsInPastEnabled()).willReturn(true);
+
+        LocalDateTime correctEndDate = LocalDateTime.of(2050, 6, 5, 11, 0);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(parseLong(CASE_ID))
+            .data(Map.of("hearingStartDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingEndDate", LocalDateTime.of(2001, 6, 5, 11, 0),
+                "hearingEndDateConfirmation", correctEndDate,
+                "hearingDateConfirmation","No"))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "hearing-in-past");
+
+        Map<String, Object> responseData = callbackResponse.getData();
+
+        assertThat(responseData.get("hearingStartDate").equals(LocalDateTime.of(2001, 6, 5, 11, 0)));
+        assertThat(responseData.get("hearingEndDate").equals(correctEndDate));
+    }
```

Comment:
Is this test necessary? Looks like you covered in test above no?

Reply:
so test above was to cover scenario of two past dates, startDate + endDate being added.
second scenario (this test) was to cover one past date, end date being added and to verify that start date remained same as initial start date added.